pkg/synopsysctl/cmd_create.go

```diff
@@ -648,172 +644,116 @@ var createOpsSightNativeCmd = &cobra.Command{
 	SilenceUsage:  true,
 	SilenceErrors: true,
+	Example: "\nRequried flags for setup with external database:\n\n 	synopsysctl create polaris native --namespace 'onprem' --version '2020.04' --gcp-service-account-path '<PATH>/gcp-service-account-file.json' --coverity-license-path '<PATH>/coverity-license-file.xml' --fqdn 'example.polaris.com' --smtp-host 'example.smtp.com' --smtp-port 25 --smtp-username 'example' --smtp-password 'example' --smtp-sender-email 'example.email.com' --postgres-host 'example.postgres.com' --postgres-port 5432 --postgres-username 'example' --postgres-password 'example' ",
 	Args: func(cmd *cobra.Command, args []string) error {
 		// Check the Number of Arguments
 		if len(args) != 0 {
 			cmd.Help()
-			return fmt.Errorf("this command takes 0 arguments")
-		}
-		if err := polarisPostgresCheck(cmd.Flags()); err != nil {
-			return err
-		}
-		return nil
-	},
-	PreRunE: func(cmd *cobra.Command, args []string) error {
-		err := createPolarisCobraHelper.SetPredefinedCRSpec("")
-		if err != nil {
-			cmd.Help()
-			return err
+			return fmt.Errorf("this command takes 0 argument, but got %+v", args)
 		}
-		cobra.MarkFlagRequired(cmd.Flags(), "version")
-		cobra.MarkFlagRequired(cmd.Flags(), "environment-dns")
-		cobra.MarkFlagRequired(cmd.Flags(), "postgres-password")
-
-		cobra.MarkFlagRequired(cmd.Flags(), "smtp-host")
-		cobra.MarkFlagRequired(cmd.Flags(), "smtp-port")
-		cobra.MarkFlagRequired(cmd.Flags(), "smtp-username")
-		cobra.MarkFlagRequired(cmd.Flags(), "smtp-password")
-		cobra.MarkFlagRequired(cmd.Flags(), "smtp-sender-email")
-
-		cobra.MarkFlagRequired(cmd.Flags(), "organization-description")
-		cobra.MarkFlagRequired(cmd.Flags(), "organization-admin-name")
-		cobra.MarkFlagRequired(cmd.Flags(), "organization-admin-username")
-		cobra.MarkFlagRequired(cmd.Flags(), "organization-admin-email")
-		cobra.MarkFlagRequired(cmd.Flags(), "polaris-license-path")
-		cobra.MarkFlagRequired(cmd.Flags(), "coverity-license-path")
-		cobra.MarkFlagRequired(cmd.Flags(), "gcp-service-account-path")
-
 		return nil
 	},
 	RunE: func(cmd *cobra.Command, args []string) error {
-		polarisObj, err := updatePolarisSpecWithFlags(cmd, namespace)
+		// Get the flags to set Helm values
+		helmValuesMap, err := createPolarisCobraHelper.GenerateHelmFlagsFromCobraFlags(cmd.Flags())
 		if err != nil {
 			return err
 		}
 
-		if err := CheckVersionExists(baseURL, polarisObj.Version); err != nil {
-			return err
+		// Update the Helm Chart Location
+		chartLocationFlag := cmd.Flag("chart-location-path")
+		if chartLocationFlag.Changed {
+			polarisReportingChartRepository = chartLocationFlag.Value.String()
+		} else {
+			versionFlag := cmd.Flag("version")
+			if versionFlag.Changed {
+				polarisReportingChartRepository = fmt.Sprintf("%s/charts/polaris-helmchart-reporting-%s.tgz", baseChartRepository, versionFlag.Value.String())
```

Comment:
This should be the polarisChartRepository not polarisReportingChartRepository

Reply:
It should also be %s/charts/polaris-helmchart-%s.tgz

Reply:
yup fixing the native mode

Reply:
done